### DISCUSSION TITLE
I Want Bleak Library to Work Integrated with Pyscirpt

### DISCUSSION THREAD

**Discussion by omerkars78** [pyscript/pyscript#1287] [Announcements]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other feature requests and couldn't find a duplicate (including also the ``type-feature`` tag)
- [X] I confirmed that it's not related to another project area (see the above section)

### What is the idea?

The Bluetooth data of the Surrounding Devices cannot be received via the web. In other words, very limited data can be obtained. However, it is almost impossible to get data such as (rssi, major, minor, uuid, deviceList) on the client side. Chorome has its own bluetooth library on this subject, but most of the methods in this library cannot work. Mobile application is required for Bletooth projects.
With the Python bleak library, the bluetooth data of the surrounding devices can be obtained very easily (((rssi,major,minor,uuid,deviceList)). If you make the "Bleak" library integrated with pyscirpt. Web-Bluetooth projects can be developed easily.

### Why is this needed

There is a project that we are currently developing, in this project, we need the bluetooth-related data of the devices with bluetooth service turned on in the vicinity. We want to do this as a web project. So it has to work on the client side. In order to do this, pyscript and bleak library need to work together. If this is possible, many web-blueooth projects will be developed around the world.

### What should happen?

Python "Bleak" library should work with "PyScript" library

### Additional Context

[https://github.com/hbldh/bleak] => this is repository of the bleak library.

**Comment by FabioRosado** [comment]
Pyscript doesn't maintain specific packages, this is work done by pyodide, you may want to read the install docs here: https://pyodide.org/en/stable/usage/faq.html#micropip-can-t-find-a-pure-python-wheel 

You have two options to get bleak to work:
- Port the library using [creating a package](https://pyodide.org/en/stable/development/new-packages.html#new-packages)
- Build the wheel yourself and use it by specifying `packages = ['your-wheel.wh']` in the py-config tag

Hope this helps 😄

**Comment by omerkars78** [comment]
Thank you very much! I will try!

Fábio Rosado ***@***.***>, 7 Ara 2022 Çar, 17:25 tarihinde
şunu yazdı:

**Reply by bilogic** [reply]
did you have any luck running bleak in pyscript?
